crates/ark/src/modules/positron/options.R

```diff
@@ -29,20 +29,19 @@ options(device = function() {
 # Set cran mirror
 repos <- getOption("repos")
 rstudio_cran <- "https://cran.rstudio.com/"
-
 if (is.null(repos) || !is.character(repos)) {
-    options(repos = c(CRAN = rstudio_cran))
+    repos <- c(CRAN = rstudio_cran)
 } else {
     if ("CRAN" %in% names(repos)) {
         if (identical(repos[["CRAN"]], "@CRAN@")) {
             repos[["CRAN"]] <- rstudio_cran
-            options(repos = repos)
         }
     } else {
         repos <- c(CRAN = rstudio_cran, repos)
-        options(repos = repos)
     }
 }
+attr(repos, "Positron") <- TRUE
+options(repos = repos)
 
 # Show Plumber apps in the viewer
```

Comment:
I have one other larger worry about this in general.

None of this is going to kick in if the user is using a version of R installed with Rig. That's because Rig sets `repos` for you in the special `/library/base/R/Rprofile` that R forcibly loads on every startup (like, rig straight up tweaks this file that you get when you download R itself)

That means that every time you load up a rig installed version of R, it will _always_ have `repos` set and it will never be `"@CRAN@"`.

https://github.com/r-lib/rig/blob/140115c9b565167670cfc6f303e6c968c563db98/src/windows.rs#L342
https://github.com/r-lib/rig/blob/140115c9b565167670cfc6f303e6c968c563db98/src/macos.rs#L788
https://github.com/r-lib/rig/blob/140115c9b565167670cfc6f303e6c968c563db98/src/linux.rs#L579-L613

That means we would _never_ add the `"Positron"` attribute, and RStudio never adds the `"RStudio"` attribute.

I can confirm this by loading up my non-rig R-devel version in a plain terminal:

```
> getOption("repos")
    CRAN
"@CRAN@"
```

vs a rig installed R 4.4

```
> getOption("repos")
                         CRAN
"https://cloud.r-project.org"
```

Now, this _may_ be ok for Positron because on Linux you can see in the above link that rig sets up PPM as the `repos` option, so that should mean that users get binaries when using renv + rig on Linux, even though renv isn't the one setting the `repos` option.

The bigger issue would be for renv itself, because it means that renv will never detect a "default" case and can't set the `repos` option to the user's chosen configuration option for `ppm.url()`, which defaults to normal PPM but can be customized by the user:
https://github.com/rstudio/renv/blob/fcfe748225a88bf56b1964868604353c2e73bf7c/R/init.R#L341

Reply:
I do not think we should let this stop us from moving on this PR, but @kevinushey may want to take a closer look at this weird rig + renv intersection issue, especially the inability to set a user customized `config$ppm.url()`

Reply:
I am not a rig user 😆 so I did not notice this.

I will add a note for renv when I say this new attribute is available.